pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -18,191 +18,144 @@ package ensurer
-// can be used to ensure a set of suggested FlowSchema configuration objects.
-func NewSuggestedFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface, lister flowcontrollisters.FlowSchemaLister) FlowSchemaEnsurer {
-	wrapper := &flowSchemaWrapper{
-		client: client,
-		lister: lister,
-	}
-	return &fsEnsurer{
-		strategy: newSuggestedEnsureStrategy(wrapper),
-		wrapper:  wrapper,
-	}
+func (*flowSchemaClient) typeName() string {
+	return "FlowSchema"
 }
 
-// NewMandatoryFlowSchemaEnsurer returns a FlowSchemaEnsurer instance that
-// can be used to ensure a set of mandatory FlowSchema configuration objects.
-func NewMandatoryFlowSchemaEnsurer(client flowcontrolclient.FlowSchemaInterface, lister flowcontrollisters.FlowSchemaLister) FlowSchemaEnsurer {
-	wrapper := &flowSchemaWrapper{
-		client: client,
-		lister: lister,
-	}
-	return &fsEnsurer{
-		strategy: newMandatoryEnsureStrategy(wrapper),
-		wrapper:  wrapper,
-	}
+func (boots *bootstrapFlowSchemas) len() int {
+	return len(boots.boots)
 }
 
-// NewFlowSchemaRemover returns a FlowSchemaRemover instance that
-// can be used to remove a set of FlowSchema configuration objects.
-func NewFlowSchemaRemover(client flowcontrolclient.FlowSchemaInterface, lister flowcontrollisters.FlowSchemaLister) FlowSchemaRemover {
-	return &fsEnsurer{
-		wrapper: &flowSchemaWrapper{
-			client: client,
-			lister: lister,
-		},
+func (boots *bootstrapFlowSchemas) get(i int) bootstrapObject {
+	return &bootstrapFlowSchema{
+		flowSchemaClient: &boots.flowSchemaClient,
+		bootstrap:        boots.boots[i],
 	}
 }
 
-// GetFlowSchemaRemoveCandidates returns a list of FlowSchema object
-// names that are candidates for deletion from the cluster.
-// bootstrap: a set of hard coded FlowSchema configuration objects
-// kube-apiserver maintains in-memory.
-func GetFlowSchemaRemoveCandidates(lister flowcontrollisters.FlowSchemaLister, bootstrap []*flowcontrolv1beta3.FlowSchema) ([]string, error) {
-	fsList, err := lister.List(labels.Everything())
+func (boots *bootstrapFlowSchemas) getExistingObjects() ([]deletable, error) {
+	objs, err := boots.lister.List(labels.Everything())
 	if err != nil {
-		return nil, fmt.Errorf("failed to list FlowSchema - %w", err)
+		return nil, fmt.Errorf("failed to list FlowSchema objects - %w", err)
 	}
-
-	bootstrapNames := sets.String{}
-	for i := range bootstrap {
-		bootstrapNames.Insert(bootstrap[i].GetName())
-	}
-
-	currentObjects := make([]metav1.Object, len(fsList))
-	for i := range fsList {
-		currentObjects[i] = fsList[i]
+	dels := make([]deletable, len(objs))
+	for i, obj := range objs {
+		dels[i] = &deletableFlowSchema{
+			FlowSchema: obj,
+			client:     boots.client,
+		}
 	}
-
-	return getDanglingBootstrapObjectNames(bootstrapNames, currentObjects), nil
+	return dels, nil
 }
 
-type fsEnsurer struct {
-	strategy ensureStrategy
-	wrapper  configurationWrapper
+type bootstrapFlowSchema struct {
+	*flowSchemaClient
```

Comment:
same here?

Reply:
We could but what would be the benefit? The costs would include writing more methods because of promotions that are no longer there.